stateresolutionv2.go

```diff
@@ -231,13 +231,16 @@ func ReverseTopologicalOrdering(input []*Event, order TopologicalOrder) []*Event
 func HeaderedReverseTopologicalOrdering(events []*HeaderedEvent, order TopologicalOrder) []*HeaderedEvent {
 	r := stateResolverV2{}
 	input := make([]*Event, len(events))
+	hisVis := make(map[string]HistoryVisibility)
```

Comment:
We know the capacity here ahead of time as `len(events)` so please give that to `make()` to reduce the number of allocations.